### PR TITLE
Add basic user registration

### LINKED ISSUE #6: Basic user registration
## Requirements
A basic user registration form that allows new people to sign up for Geek.Zone membership. This initial basic form should focus on the minimum required to register someone to Geek.Zone.

* full name
* preferred name
* email address
* date of birth (saved to the db in iso8601 format)
* a password (so they can log in to their account)
* Their agreement to the geek.zone constitution

This first pass does not include taking payments

### PR DESCRIPTION
**Pull request by CarwynNelson**
This pull request introduces some very basic user registration for membership management. Little effort has been put into validation and no effort has been put in to styling - we can build on these in further commits :)

The form can be accessed at `/memberships/register` and redirects a user to `/memberships/thanks` when registration is successful. We are capturing the data needed to consider someone a Geek.Zone member (excluding payment). The data required is:
- full name
- preferred name
- email address
- date of birth (saved to the db in iso8601 format)
- a password (so they can log in to their account)
- Their agreement to the geek.zone constitution

Some interesting extra info:
- The Member model is associated with the build in django User Model which can deal with auth, password storage, etc for us
- The preferred name is defaulted to the users full name if it is not entered
- The User admin page has some slight modifications to it that hide certain fields that we store on the Member model, and make some date fields read only

Resolves #6

### CODE REVIEW

**Comment by CarwynNelson** [comment]
Being a good citizen and all that I should probably write some automated tests before we merge this in (I have set to do this in Django).

Also I'm considering setting up a basic RDS database and adding migrations to the build pipeline before we merge this so that it immediately works on `test.geek.zone`

**Comment by CarwynNelson** [comment]
I have written some basic tests that I think are useful but not too onerous. Now that #20 has been merged in this should be able to safely go to live and "just work".

File: memberships/tests.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        self.assertFormError(
+            response, "form", "constitution_agreed", "This field is required."
+        )
+
+    def test_preferred_name_defaults_to_full_name(self):
+        self.client.post(
+            reverse("register"),
+            {
+                "full_name": "test person",
+                "email": "test@person.com",
```
Comment (jamesgeddes) [inline memberships/tests.py:26]: Being really picky but can we test with example.com please? We know that will never resolve to anything other than the example site, unlike this particular domain.

Comment (CarwynNelson) [reply memberships/tests.py]: Sorted.

**Pull request merged**